src/lib/utils/ct_utils.h

```diff
@@ -107,6 +107,54 @@ constexpr inline T value_barrier(T x) {
    }
 }
 
+/**
+* A Choice is used for constant-time conditionals.
+*
+* Internally it always is either all 0 bits or all 1 bits.
+*/
+class Choice final {
+   public:
+      /**
+      * If v == 0 return an unset (false) Choice, otherwise a set Choice
+      */
+      template <typename T>
+      constexpr static Choice from_int(T v)
+         requires(std::is_unsigned<T>::value)
+      {
+         const auto v_is_0 = ct_is_zero<T>(value_barrier<T>(v));
+         return Choice(ct_is_zero<uint32_t>(static_cast<uint32_t>(v_is_0)));
+      }
+
+      template <typename T>
+      constexpr T choose(T a, T b)
+         requires(std::is_unsigned<T>::value)
+      {}
```

Comment:
```suggestion
      template <std::unsigned_integral T>
      constexpr T choose(T a, T b)
      {}
```

... also: this is lacking an implementation (and perhaps a docstring: what is chosen over what?).

Reply:
This is stray - I decided to go with converting Choice to a Mask for selecting a T but forgot to remove this

Reply:
Removed